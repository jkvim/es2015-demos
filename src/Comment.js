import React, { Component } from 'react'
import PropTypes from 'prop-types';

import style from './Comment.scss';

class Comment extends Component {
    static propTypes = {
        username: PropTypes.string,
        content: PropTypes.string
    }
    
    render() {
        const { username, content } = this.props
        return (
            <div className={style.comment}>
                <p className={style.username}>{username}</p>:
                <p className={style.content}>{content}</p>
            </div>
        )
    }
}

export default Comment;
