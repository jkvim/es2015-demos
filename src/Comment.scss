.comment {
  display: flex;
  align-items: center;
  border-bottom: 1px solid #EEE;
}

.username {
  color: #00a3cf;
  padding: 5px;
  font-style: italic;
}