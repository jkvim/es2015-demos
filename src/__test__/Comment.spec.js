import React from 'react';
import renderer from 'react-test-renderer';

import Comment from '../Comment';


describe('<Comment /> snapshot', () => {
    it('should render correctly', () => {
        const tree = renderer
            .create(<Comment username="XiaoMing" content="Hello World" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
