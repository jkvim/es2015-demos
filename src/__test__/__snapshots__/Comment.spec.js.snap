// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> snapshot should render correctly 1`] = `
<div
  className="comment"
>
  <p
    className="username"
  >
    XiaoMing
  </p>
  <p
    className="divider"
  >
    :
  </p>
  <p
    className="content"
  >
    Hello World
  </p>
</div>
`;
